File: voiture-resevation-frontend/src/app/controller/validators/client.validator.ts
import {Validator, ValidatorItem} from "@bshg/validation";
import {Client} from "src/app/controller/entities/client";

export class ClientValidator extends Validator<Client> {
  nom = new ValidatorItem<string>(
    () => this.item().nom,
    (value) => this.item().nom = value,
    (value) => {
      this.nom.stringValidators
        ?.required()
        ?.valid()
    }
  )
  prenom = new ValidatorItem<string>(
    () => this.item().prenom,
    (value) => this.item().prenom = value,
    (value) => {
      this.prenom.stringValidators
        ?.required()
        ?.valid()
    }
  )
  age = new ValidatorItem<number>(
    () => this.item().age,
    (value) => this.item().age = value,
    (value) => {
      this.age.numberValidators
        ?.required()
        ?.valid()
    }
  )
  localite = new ValidatorItem<string>(
    () => this.item().localite,
    (value) => this.item().localite = value,
    (value) => {
      this.localite.stringValidators
        ?.required()
        ?.valid()
    }
  )
  adresse = new ValidatorItem<string>(
    () => this.item().adresse,
    (value) => this.item().adresse = value,
    (value) => {
      this.adresse.stringValidators
        ?.required()
        ?.valid()
    }
  )
  datePromis = new ValidatorItem<Date>(
    () => this.item().datePromis,
    (value) => this.item().datePromis = value,
    (value) => {
      this.datePromis.validators
        ?.required()
        ?.valid()
    }
  )

  override validatorItems: ValidatorItem<any>[] = [
    this.nom,
    this.prenom,
    this.age,
    this.localite,
    this.adresse,
    this.datePromis,
  ]

  getValidatorItem(fieldName: string): ValidatorItem<any> {
    // @ts-ignore
    return this[fieldName];
  }

  static init(item: () => Client): ClientValidator {
    const validator = new ClientValidator()
    validator.item = item
    return validator;
  }

}

File: voiture-resevation-backend/src/main/java/com/example/voitureresevation/entity/validators/ClientValidator.java
package com.example.voitureresevation.entity.validators;

import com.example.voitureresevation.entity.Client;
import com.example.voitureresevation.zutils.validators.Validator;
import com.example.voitureresevation.zutils.validators.ValidatorItem;

import java.time.LocalDate;
import java.util.List;

public class ClientValidator extends Validator<Client> {
    public ClientValidator(Client item) {
        super(item);
    }

    public static void validate(Client item) {
        new ClientValidator(item).validate();
    }

    public ValidatorItem<String> nom = new ValidatorItem<>(
            "nom",
            () -> this.getItem().getNom(),
            (String value) -> {
                this.nom.getValidators()
                        .required()
                        .valid();
            }
    );
    public ValidatorItem<String> prenom = new ValidatorItem<>(
            "prenom",
            () -> this.getItem().getPrenom(),
            (String value) -> {
                this.prenom.getValidators()
                        .required()
                        .valid();
            }
    );
    public ValidatorItem<Integer> age = new ValidatorItem<>(
            "age",
            () -> this.getItem().getAge(),
            (Integer value) -> {
                this.age.getValidators()
                        .required()
                        .valid();
            }
    );
    public ValidatorItem<String> localite = new ValidatorItem<>(
            "localite",
            () -> this.getItem().getLocalite(),
            (String value) -> {
                this.localite.getValidators()
                        .required()
                        .valid();
            }
    );
    public ValidatorItem<String> adresse = new ValidatorItem<>(
            "adresse",
            () -> this.getItem().getAdresse(),
            (String value) -> {
                this.adresse.getValidators()
                        .required()
                        .valid();
            }
    );
    public ValidatorItem<LocalDate> datePromis = new ValidatorItem<>(
            "datePromis",
            () -> this.getItem().getDatePromis(),
            (LocalDate value) -> {
                this.datePromis.getValidators()
                        .required()
                        .valid();
            }
    );

    @Override
    public List<ValidatorItem<?>> validatorItems() {
        return List.of(
                this.nom,
                this.prenom,
                this.age,
                this.localite,
                this.adresse,
                this.datePromis
        );
    }
}
